src/backup.cpp

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2020 YADRO
+
+#include "accounts.hpp"
+#include "backup.hpp"
+#include "manifest.hpp"
+
+#include <vector>
+
+namespace fs = std::filesystem;
+
+// clang-format off
+/** @brief List of base configuration files/directories. */
+static const std::vector<const char*> baseConfigs = {
+    "etc/dropbear/dropbear_rsa_host_key",
+    "etc/ipmi_pass",
+    "etc/machine-id",
+};
+
+/** @brief List of network configuration files/directories. */
+static const std::vector<const char*> networkConfigs = {
+    "etc/hostname",
+    "etc/systemd/network",
+    "var/lib/first-boot-set-hostname",
+};
+// clang-format on
+
+Backup::~Backup()
+{
+    if (!tmpDir.empty())
+    {
+        fs::remove_all(tmpDir);
+    }
+}
+
+void Backup::backup()
+{
+    if (fs::exists(archiveFile))
+    {
+        std::string err = "Backup file already exists: ";
+        err += archiveFile;
+        throw std::runtime_error(err);
```

Comment:
Don't you want to create your own exception with a printf-style format string and use it instead of the series of string concatenations? ;)

Reply:
The sky is the limit :) Let's leave it for the future.